### PR TITLE
KAS-1170: Modals upgrade

### PR DESCRIPTION
**Pull request by rafael-mindreau**
image

**WHAT**
Changed all modals to reflect the new version (as seen in the sample above)

The modal upgrade concerns the following three main conceptual changes:
* The top and bottom are now fixed in place
* The modal may grow to a max height of 95% of the viewport
* The middle content overflows with a scroll automatically when this max height is reached

Some common bugs this introduced (and could still introduce):
* Missing padding around the footer - since the footer is custom for most dialogs
* Scroll wrappers made middle content scrollable before reaching its max height
* In some cases the modal needed to be larger (used the large modifier)

### CODE REVIEW

File: app/styles/govflanders/components/_modal.scss [hunk 1]
```diff
@@ -97,3 +105,9 @@ $modal-content-padding: 1.5rem;
     }
   }
 }
+
+.vl-modal-dialog__content {
+  .data-table-pagination {
+    margin-bottom: 0em;
```
Comment (Wolfr) [inline app/styles/govflanders/components/_modal.scss]: 0 values should be unitless. This is a new style rule, I will add it to Ygdir.

File: app/styles/govflanders/components/_modal.scss [hunk 2]
```diff
@@ -12,10 +12,18 @@ $modal-content-padding: 1.5rem;
   .vl-modal__backdrop {
     @include backdrop;
   }
+
+  .vl-modal__backdrop {
+    z-index: 52;
+  }
+
+  tbody {
+    background-color:white;
```
Comment (Wolfr) [inline app/styles/govflanders/components/_modal.scss]: Code style issue - put a space after the semicolon.

File: app/styles/app.scss [hunk 3]
```diff
@@ -175,19 +169,6 @@ tr.disabled {
...
-  }
-}
-
-.vl-modal-dialog__content {
-  padding: 2rem 0;
-}
-
 .max-width {
   text-overflow: ellipsis;
   max-width: 800px;
```
Comment (Wolfr) [inline app/styles/app.scss]: Code issue - usage of px - Ygdir standard 9 - Use rems

File: app/styles/app.scss [hunk 4]
```diff
@@ -159,12 +159,6 @@ tr.disabled {
...
 
-.vl-modal-dialog {
-  display: block;
-  z-index: 40;
-  width: 80rem;
-}
-
 .vlc-scroll-wrapper__body.full-height {
   > .ember-view {
     height: 100% !important;
```
Comment (Wolfr) [inline app/styles/app.scss]: Should check if it's really needed to use height 100% here. This seems a bit unnecessary.

**Comment by Wolfr** [comment]
Screenshot 2020-01-16 at 14 44 14

Bug: de modal komt onder de hoofdbalk.

**Comment by rafael-mindreau** [comment]
> Screenshot 2020-01-16 at 14 44 14
> 
> Bug: de modal komt onder de hoofdbalk.

@Wolfr Couldn't reply to the comment with the image for some reason 🤔 I can either fix that with z-index, or make sure the modal's max height doesn't clip with the top header. I'd say I'm in favour of the first solution, but that would make it overlap with the header (kind of weird as well). Your thoughts?

File: app/styles/app.scss (hunk 4 above)
Comment (rafael-mindreau) [reply app/styles/app.scss]: This code was not introduced by me, and subsequently I will not expand beyond the scope of the issue.

File: app/styles/app.scss (hunk 3 above)
Comment (rafael-mindreau) [reply app/styles/app.scss]: This code was not introduced by me, and subsequently I will not expand beyond the scope of the issue.

File: app/styles/govflanders/components/_modal.scss (hunk 2 above)
Comment (rafael-mindreau) [reply app/styles/govflanders/components/_modal.scss]: I copied/refactored this code over from the main `app.scss`, but here the boyscout rule applies, so I will make the changes.

File: app/styles/govflanders/components/_modal.scss (hunk 1 above)
Comment (rafael-mindreau) [reply app/styles/govflanders/components/_modal.scss]: I copied/refactored this code over from the main `app.scss`, but here the boyscout rule applies, so I will make the changes.

**Comment by rafael-mindreau** [comment]
> > Screenshot 2020-01-16 at 14 44 14
> > Bug: de modal komt onder de hoofdbalk.
> 
> @Wolfr Couldn't reply to the comment with the image for some reason 🤔 I can either fix that with z-index, or make sure the modal's max height doesn't clip with the top header. I'd say I'm in favour of the first solution, but that would make it overlap with the header (kind of weird as well). Your thoughts?

image

I came up with a solution for not having it clip with the govflanders nav widget for which we cannot control the z-index cleanly anyway.

**Comment by rafael-mindreau** [comment]
@Wolfr deze had een `z-index` van 999, maar scoped in zijn eigen context, ik geraakte er niet boven 🤔 Zouden we best voorlopig met die tussenoplossing zo laten? Is natuurlijk wel ambetant!

**Comment by Wolfr** [comment]
Niks aan te doen, er is een algemene discussie over deze header, dit is
argument #5 of #6 om die externe header weg te halen. Dank om te proberen
alleszins.

**Comment by Wolfr** [comment]
Screenshot 2020-01-17 at 10 40 17

Nog een probleempje gevonden in deze modal, als je een dossier verwijderd. Hiervoor moet je op de ••• more knop klikken in de dossiers datagrid.

**Comment by Wolfr** [comment]
Als ik escape druk in deze modal doet dat ook nog niks, ik zag nochtans een vermelding van "on test" https://kanselarij.atlassian.net/browse/KAS-1042

**Comment by Joachimzeelmaekers** [comment]
Je moet eerst eens in de modal klikken.
Roel is daar aan bezig.

On 17 Jan 2020, at 10:43, Wolfr <notifications@github.com<mailto:notifications@github.com>> wrote:


Als ik escape druk in deze modal doet dat ook nog niks, ik zag nochtans een vermelding van "on test" https://kanselarij.atlassian.net/browse/KAS-1042

**Comment by RoelGo** [comment]
Bovenaan de modals van instellingen is er een klein artifactje als je de modal selecteert

Screenshot 2020-01-17 at 10 51 36

**Pull request merged**